src/overlays/actors/ovl_Obj_Y2shutter/z_obj_y2shutter.c

```diff
@@ -28,24 +41,136 @@ const ActorInit Obj_Y2shutter_InitVars = {
     (ActorFunc)ObjY2shutter_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9AEB0[] = {
+ShutterInfo sShutterInfo[] = {
+    // rectangular shutter door, activated by floor plate
+    { object_kaizoku_obj_DL_0032A0, &object_kaizoku_obj_Colheader_0035B0, 120.0f, 20.0f, 3.0f, -20.0f, 3.0f, 4, 8,
+      160 },
+    // hexagonal shutter, activated by hitting crystal switch
+    { object_kaizoku_obj_DL_005720, &object_kaizoku_obj_Colheader_005EC8, 150.0f, 1.0f, 0.04f, -1.0f, 0.04f, 6, 0xC,
+      160 },
```

Comment:
Might as well make an enum for the type and name these assets rather than leaving them with comments. The activation's also just triggered by flags, so precisely how is not an inherent property of this actor.